bgpd/bgp_route.c

```diff
@@ -2333,6 +2333,18 @@ void bgp_best_selection(struct bgp *bgp, struct bgp_dest *dest,
 	return;
 }
 
+bool bgp_check_advertise(struct bgp *bgp, struct bgp_dest *dest)
+{
+	bool advertise = true;
+
+	if (BGP_SUPPRESS_FIB_ENABLED(bgp) &&
+	    CHECK_FLAG(dest->flags, BGP_NODE_FIB_INSTALL_PENDING) &&
+	    (!bgp_option_check(BGP_OPT_NO_FIB)))
```

Comment:
You could cut this whole function to one line by simply returning this condition expression